R/zchunk_LA161.Cstorage_china.R

```diff
@@ -0,0 +1,125 @@
+      mutate(Cumul_MtC = cumsum(MtC))
+
+    L161.Cstorage_province <- province_data %>%
+      # use "type=1" to indicate that we want to return only exact matches to the data in the table.
+      # only take the (0, 0.2, 0.4, 0.6, 0.8, 1) points in the quartiles
+      filter(Cumul_MtC %in% quantile(Cumul_MtC, c(0, 0.2, 0.4, 0.6, 0.8, 1), type = 1)) %>%
+      group_by(province.name) %>%
+      mutate(# this will create grade equal to grade 1 for 0, grade 2 for 0.2, etc.
+             grade = paste("grade",1:6, sep = " "),
+             # from the cumulative totals, return to the marginal quantities associated with each grade
+             MtC = Cumul_MtC- lag(Cumul_MtC,1,0)) %>%
+      ungroup %>%
+      # Return from the list to a data frame with all necessary data
+      select(province.name, grade, Cost_1990USDtC, MtC) %>%
+      # Add cost grades to provinces without data
+      bind_rows(CCS_China_add) %>%
+      mutate(# Setting a minimum cost of 0 on CO2 storage and transport projects
+             Cost_1990USDtC = pmax(Cost_1990USDtC, 0))
+
+    # Manually changing cost curves to avoid grades 2 or 3 being 0. Value determined by linear interpolation. If the data changes in the future, will need to change this
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Jilin" & L161.Cstorage_province$grade== "grade 2")]    <- 4.625
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Xinjiang" & L161.Cstorage_province$grade== "grade 2")] <- 4.3
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Yunnan" & L161.Cstorage_province$grade== "grade 2")]   <- 5.74
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Yunnan" & L161.Cstorage_province$grade== "grade 3")]   <- 11.48
+
+    #Extending the plateau (bin 2) of the cost curve using additional resource data
+    L161.Cstorage_province_g2 <- L161.Cstorage_province %>%
+      filter(grade == "grade 2") ->
+      L161.Cstorage_province_g2
+
+    L161.Cstorage_province_g2 %>%
+      mutate(add_storage = left_join_error_no_match(L161.Cstorage_province_g2, Unallocated_CStorage, by = c("province.name" = "province.name"))$MtCO2,
+             #Convert to MtC for additional resource
+             add_storage = add_storage * (1 / emissions.CONV_C_CO2)) ->
+      L161.Cstorage_province_g2
+
+    #Gansu is a special case, we are not adding storage
+    L161.Cstorage_province_g2$add_storage[which(L161.Cstorage_province_g2$province.name=="Gansu")] <- 0
+
+    L161.Cstorage_province_g2 %>%
+      transform(MtC=as.numeric(MtC)) %>%
+      mutate(MtC = MtC + add_storage) %>%
+      select(province.name, grade, Cost_1990USDtC, MtC) ->
+      L161.Cstorage_province_g2
+
+    L161.Cstorage_province %>%
+      filter(grade !="grade 2") %>%
+      bind_rows(L161.Cstorage_province_g2) %>%
+      mutate(province.name_0 = province.name) %>%
```

Comment:
??

Reply:
What happens here:
Add a new column whose value is the abbreviation of province name

I want to use the function province_names_mappings to get the abbreviation of province name, but this function is replacing province name with the abbreviation. Output needs both province name and the abbreviation.

So I add a new column whose value is same as province.name, then using province_names_mappings to get the abbreviation, finally rename the added column to make the output have both province name and the abbreviation.

Reply:
See my revisions. The function has the option that does not drop the province name.